Custom system fixes

zonkmachine opened the pull request:
Fixes https://github.com/pioneerspacesim/pioneer/issues/5581

* Updates lua system files in **/data/systems/custom** to json
* Fixes bodies with same gravity or density
* Fixes planets with unrealistic stats such as, average densities larger than the densest known base elements or zero mass.
* Updates values in Sol. **Aega** (moon of Jupiter) is removed.
* Save 01_epsilon_eridani.lua as .json with fixed values.
* All lua files are gone

zonkmachine commented:
> * The only system left to fix is essentially Epsilon Eridani. I don't have a fix for 00_sol.lua and 01_epsilon_eridani.lua so it may have to be fixed in a later PR.

What I meant is: I don't have a fix for the need for keeping the lua version of a file in **data/systems** when saving it as **.json** in **data/systems/custom**. In any case, I've fixed up the final values in **01_epsilon_eridani.lua** and saved it as .json. The original **.lua** remains. There should be no more wonky values in the universe.

impaktor commented:
Now there's two systems with `01...` in the file name in the same folder, I assume this is not a problem, but still... could just move the Werribee to be prefixed with N+1 instead?

The issues you've discovered - would they indicate there are checks that aught to be added to the system generation to exclude certain areas of parameter space? Perhaps @sturnclaw would want those codified "If X and Y then Z must be less than Q", or something like that?

zonkmachine commented:
> so to migrate Sol to JSON, you also need to migrate Epsilon Eridani and Wolf 359.

I don't get this. I convert these systems to json and remove the lua counterpart. Isn't that migrating to json? I am migrating Epsilon Eridani and Wolf 359 to json in this PR but removing the lua files fail.

zonkmachine commented:
All todo's are fixed and the branch has been rebased on top of master. Systems have been resaved with the updated editor. Looks fine. Only systems that changed on saving again was 06_61virginis which flipped the values of the atmosphere of a gas giant. Noting I could see on inspection on site (from low orbit). Rondel also looks the same but the diff on resaving it showed a lot of changes. I don't know why Rondel was different from the rest.

I'll test it some more.

zonkmachine commented:
> Because the contents of 04_rondel.lua are "empty" (see below)

Yes. One more time. When I first opened 04_rondel.lua in the editor and saved it as json, it was depending on the random seed. Then I rebased the PR on master and doing so I once more opened the systems, rondel now not random, in the editor and saved them. All systems were saved with only the file ending fixed, except 04_rondel.json.

zonkmachine commented:
@sturnclaw Now that it's reproducible, perhaps I can open a separate issue about it instead (with clearer instructions how to reproduce it) and I can rebase/squash this PR and maybe it's a merge?

zonkmachine commented:
Needs more testing. Wait with merging.

zonkmachine committed:
Update custom system files

- Updates lua system files in /data/systems/custom to json.
- Fixes bodies with same gravity or density.
- Fixes planets with unrealistic stats such as average densities,
  larger than the densest known base elements, or zero mass.
- Updates values in Sol. Aega (moon of Jupiter) is removed.
- Give Torvalds station a proper seed value.
- Save 01_epsilon_eridani.lua as .json with fixed values. The original
  lua file is kept, similarly to 00_sol.lua which is also doubled in
  data/systems/custom.

zonkmachine commented:
> @zonkmachine what's the status on the PR? I see a force-push.

I noticed that Itzalean had changed so I had to take a closer look. I'll have to retrace my steps to find out how it happened but for now the fix was to redo all the systems (open/save in the editor) apart from Sol. Did that and force pushed. It could very well have been an error on my behalf and I should be able to find out a way to repeat it. I don't know if other systems was affected. It looks good now but it would probably be a good idea to give me some time to trouble shoot and see how it happened.

sturnclaw commented:
@zonkmachine any updates on this PR's status? Is it ready for merge? I see it's still marked as draft...

sturnclaw commented:
@zonkmachine if you're interested in this PR being included in the upcoming release, I'd ask that you rebase it and mark ready for review by Jan 21st. I'd certainly like to have it included!

impaktor commented:
>  I'd ask that you rebase it and mark ready for review by Jan 21st. I'd certainly like to have it included!

@zonkmachine PING PING PINGa DING DING DING

zonkmachine commented:
@impaktor @sturnclaw Sorry for the late reply. I'm slowly getting back to coding.
After the custom systems were given saner values the hills around Itzalean disappeared. They are still there but of a fraction of the original heights. I have a hunch that the terrain engine may have been tuned on bad values from before planet densities were checked, so now that those values are sane the geographic features are more modest.

zonkmachine commented:
Rebased on master. Apart from the added `factions/` directory, here are the other changes in `data/systems/` since the last update on this PR some seven months ago.

Before pushing - **git diff origin/customsystemfixes data/systems/** 
    **(data/systems/factions)** omitted.
```diff
zonkmachine@zonkmachine-VivoBook-ASUSLaptop-X513EA-K513EA:~/builds/pioneer$ git diff origin/customsystemfixes data/systems/
diff --git a/data/systems/custom/02_barnard_s_star.json b/data/systems/custom/02_barnard_s_star.json
index d7b543780..af0fe8f82 100644
--- a/data/systems/custom/02_barnard_s_star.json
+++ b/data/systems/custom/02_barnard_s_star.json
@@ -293,7 +293,7 @@
         },
         {
             "agricultural": "f0/0",
-            "argOfPeriapsis": "f2/2318766477",
+            "argOfPeriapsis": "f0/1824256063",
             "aspectRatio": "f1/0",
             "atmosColor": [
                 0,
@@ -316,14 +316,14 @@
             "mass": "f0/347892",
             "metallicity": "f0/21904333",
             "name": "Name and Form",
-            "orbitalOffset": "f2/2665816616",
-            "orbitalPhase": "f1/1982595006",
+            "orbitalOffset": "f4/84002509",
+            "orbitalPhase": "f4/699841604",
             "parent": 6,
             "population": "f0/0",
             "radius": "f0/58841051",
             "rotationPeriod": "f0/4166547773",
             "rotationPhase": "f0/0",
-            "seed": 4285154954,
+            "seed": 3752954798,
             "semiMajorAxis": "f0/3440268",
             "type": "PLANET_ASTEROID",
             "volatileIces": "f100/0",
@@ -332,7 +332,7 @@
         },
         {
             "agricultural": "f0/0",
-            "argOfPeriapsis": "f5/745562530",
+            "argOfPeriapsis": "f4/1181111262",
             "aspectRatio": "f1/0",
             "atmosColor": [
                 0,
@@ -353,13 +353,13 @@
             "metallicity": "f0/0",
             "name": "OPLI Contemplation",
             "orbitalOffset": "f1/767170682",
-            "orbitalPhase": "f3/3675485893",
+            "orbitalPhase": "f4/3723924247",
             "parent": 7,
             "population": "f0/0",
             "radius": "f0/0",
             "rotationPeriod": "f0/0",
             "rotationPhase": "f0/0",
-            "seed": 1359952880,
+            "seed": 2221384235,
             "semiMajorAxis": "f0/0",
             "type": "STARPORT_SURFACE",
             "volatileIces": "f0/0",
@@ -772,20 +772,23 @@
     "comment": "",
     "faction": "Solar Federation",
     "govType": "EARTHCOLONIAL",
-    "lawlessness": "f0/0",
-    "longDesc": "Barnard's Star is a very low-mass red dwarf star.  Somewhere between 7 and 12 billion years old, it is probably one of the most ancient stars in the galaxy.  Despite that, it is still fairly active.  Pilots entering the system are warned that there might be considerable stellar activity, including flares and massive coronal ejections.\n\nOne of the first stars to be visited after the introduction of interstellar travel, Barnard's Star was found to be a much smaller system compared to Sol, with only twelve bodies. Named after the Twelve Causes of Suffering from Buddhist suttas.  Habitats were built here to serve as federal prison colonies.\n\nA permit is normally required in order to enter this system whilst carrying weapons.",
+    "index": 0,
+    "lawlessness": "f0/42949672",
+    "longDesc": "Barnard's Star is a very low-mass red dwarf star.  Somewhere between 7 and 12 billion years old, it is probably one of the most ancient stars in the galaxy.  Despite that, it is still fairly active.  Pilots entering the system are warned that there might be considerable stellar activity, including flares and massive coronal ejections.\n\nOne of the first stars to be visited after the introduction of interstellar travel, Barnard's Star was found to be a much smaller system compared to Sol, with only twelve bodies. Named after the Twelve Causes of Suffering from Buddhist suttas.  Habitats were built here to serve as Federal prison colonies.\n\nA permit is normally required in order to enter this system whilst carrying weapons.",
     "name": "Barnard's star",
+    "overrideRandom": true,
     "pos": [
         2.0799999237060547,
         0.0560000017285347,
         0.47999998927116394
     ],
+    "root": 0,
     "sector": [
         -1,
         0,
         0
     ],
-    "seed": 0,
+    "seed": 1754435159,
     "shortDesc": "Earth Federation prison colony",
     "stars": [
         "STAR_M"

```

sturnclaw commented:
Is this PR ready for review?

zonkmachine commented:
> review

I think so but I haven't had time to try it out properly after the last rebase. I would most of all recommend testing the start at Cydonia as it was changed quite a lot to remove the wonky physics. It may need some tuning.

Mc-Pain commented:
Discovered that Rhea's atmosphere in game is way too dense for actual Rhea (it's density around 1e5~1e6 molecules/cm^3)

According to ideal gas law for 81K, actual pressure is ~1e-14 atm
